src/parser/AsyncParser.js

```diff
@@ -0,0 +1,45 @@
+const parser = require('@asyncapi/parser');
+const filesystem = require('fs');
+
+/**
+ * Asynchronously Parses the provided yaml or json file .
+ *
+ *
+ * @param filepath The path of the async-api spec file.
+ * @constructor
+ */
+const AsyncParser  =  (filepath, opts) => {
+  const parserContext = this;
+  parserContext.ready = false;
+
+  async function mapAsyncApiToHandler() {
+    try {
+      parserContext.content = filesystem.readFileSync(filepath).toString();
+    } catch (err) {
+      console.log(`\nError in parsing the file. Details: ${err}`);
+    }
+    const parsed = await parser.parse(parserContext.content);
+    parserContext.ready = true;
+    parserContext.serverUrl = parsed._json.servers['production'].url;
+    parserContext.productionServerInfo = parsed.servers();
+    parserContext.PublishOperations = {};
+    parserContext.SubscribeOperations = {};
+    for (const [key,value] of Object.entries(parsed.channels())) {
+      if (value.publish())  Object.assign(parserContext.PublishOperations ,{ [key]: value.publish()._json,
+        ['plot-id']: value._json['x-plot'],
+        ['plot-group']: value._json['x-group']});
+      if (value.subscribe())  Object.assign(parserContext.SubscribeOperations ,{ [key]: value.subscribe()._json,
+        ['plot-id']: value._json['x-plot'],
+        ['plot-group']: value._json['x-group']});
+    }
+    console.log(`\nFound ${Object.keys(parserContext.PublishOperations).length} testable Operations`);
+    return parserContext;
+  }
+
+  return {
+    mapAsyncApiToHandler
+  };
```

Comment:
What's the reason for wrapping the actual parsing in another function? 🤔

Reply:
i want to able to divide the parsing process in different parts. In case of some error i want it to be handled and caught in the cli and not handled in the parser. I think that because i want all the logging to be done by one interface so that i am able to add smth on top of it in the future.(add a stream to persistently log in a file etc)

Reply:
@jonaslagoni  w8 dont merge it yet one last thing

Reply:
ok i removed it for simplicity